zrml/orderbook/src/lib.rs

```diff
@@ -298,40 +280,45 @@ mod pallet {
             maker_fill.bmul_bdiv_floor(taker_full_fill, maker_full_fill)
         }
 
-        fn ensure_ratio_quotient_valid(order_data: &OrderOf<T>) -> DispatchResult {
-            let maker_full_fill = order_data.taker_amount;
-            // this ensures that partial fills, which fill nearly the whole order, are not executed
-            // this protects the last fill happening
-            // without a division by zero for `Perquintill::from_rational`
-            let is_ratio_quotient_valid = maker_full_fill.is_zero()
-                || maker_full_fill >= T::AssetManager::minimum_balance(order_data.taker_asset);
-            ensure!(is_ratio_quotient_valid, Error::<T>::PartialFillNearFullFillNotAllowed);
-            Ok(())
+        /// The order account for a specific order id.
+        #[inline]
+        pub(crate) fn order_account(order_id: OrderId) -> AccountIdOf<T> {
+            T::PalletId::get().into_sub_account_truncating(order_id)
+        }
+
+        /// The reserve ID of the order book pallet.
+        #[inline]
+        pub(crate) fn reserve_id() -> [u8; 8] {
+            T::PalletId::get().0
         }
 
         fn do_remove_order(order_id: OrderId, who: AccountIdOf<T>) -> DispatchResult {
             let order_data = <Orders<T>>::get(order_id).ok_or(Error::<T>::OrderDoesNotExist)?;
 
             let maker = &order_data.maker;
             ensure!(who == *maker, Error::<T>::NotOrderCreator);
-
-            let missing = T::AssetManager::unreserve_named(
-                &Self::reserve_id(),
-                order_data.maker_asset,
-                maker,
-                order_data.maker_amount,
-            );
-
-            debug_assert!(
-                missing.is_zero(),
-                "Could not unreserve all of the amount. reserve_id: {:?}, asset: {:?} who: {:?}, \
-                 amount: {:?}, missing: {:?}",
-                Self::reserve_id(),
-                order_data.maker_asset,
-                maker,
-                order_data.maker_amount,
-                missing,
-            );
+            let order_account = Self::order_account(order_id);
+            let asset = order_data.maker_asset;
+            let amount = order_data.maker_amount;
+
+            if !T::AssetManager::reserved_balance(asset, &maker).is_zero() {
```

Comment:
Why not `reserved_balance_named` here instead? In that way we wouldn't have conflicts with potential other pallets.

Additionally I think the asset could be an outcome asset, which includes the market id, then it's fine, but it could also be the base asset. Other problem we have here then is, that if you have another order on the order book with the same base asset, you would go into this branch, right?

One way to solve this is by putting a more fine-grained reserve identifier onto the reserve, which directly identifies the asset additionally to the pallet identifier.

Reply:
Additionally you could also check for the amount in question. Like, we have to have at least `amount` available, otherwise it's the other case.

Reply:
Can't you just check if the asset is created using orml_tokens which has reserves? This way would be the safest approach. Because this is exactly what we want to know here. If the asset was either created by pallet_assets or orml_tokens. Maybe it's worth to break the abstraction of the asset router here in favor of clarity?

Reply:
But I don't see any case right now how this `reserved_balance_named` could fail right now so that it accidentally uses the reserve case instead of the transfer case. So I suppose it's fine.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1283/commits/5185e92e6585f2d332dc42f542de2080ffc7e1d4